yeonjae/programmers-1/햄버거만들기.js

```diff
@@ -0,0 +1,22 @@
+function solution(ingredient) {
+  const orderHamburgerIngredientArr = [];
+  let hamburgerCnt = 0;
+
+  ingredient.forEach((el) => {
+    orderHamburgerIngredientArr.push(el);
+
+    if (orderHamburgerIngredientArr.length >= 4) {
+      const find = orderHamburgerIngredientArr.slice(-4).join("");
+
+      if (find === "1231") {
+        orderHamburgerIngredientArr.pop();
+        orderHamburgerIngredientArr.pop();
+        orderHamburgerIngredientArr.pop();
+        orderHamburgerIngredientArr.pop();
+        hamburgerCnt++;
+      }
+    }
+  });
+
+  return hamburgerCnt;
```

Comment:
```suggestion
function solution(ingredient) {
  const orderHamburgerIngredientArr = [];
  let hamburgerCnt = 0;

  ingredient.forEach((el) => {
    orderHamburgerIngredientArr.push(el);

    const lastFourElement = orderHamburgerIngredientArr.slice(-4).join("");

    if (lastFourElement === "1231") {
      orderHamburgerIngredientArr.splice(-4);
      hamburgerCnt++;
    }
  });

  return hamburgerCnt;
}
```
- 풀이 잘봣씁니다 🙇‍♂️
- pop을 4개 호출하기보다 `splice(-4)`을 사용한다면 좀 더 깔끔한 풀이가 될 것 같습니다 ! 
- 또한, find라는 변수는 `동사`기 때문에 값을 의미하는 변수에는 적합하지 않은 것 같습니다. 우선 저는 lastFourElement라고 수정해줘봤씁니다 ~ 
- 아래 풀이는 제 풀이인데 reduce와 햄버거인지 체크하는 함수를 분리한 풀이입니다. 참고해보세요~!
```js
function solution(ingredient) {
  const stack = [];

  return ingredient.reduce((acc, cur) => {
    stack.push(cur);

    if (isHamburger(stack)) {
      stack.splice(-4);
      return acc + 1;
    }
    return acc;
  }, 0);
}

function isHamburger(stack) {
  const stackLength = stack.length;
  const hamburger = stack.slice(stackLength - 4, stackLength).join("");

  return hamburger === "1231";
}

```